### PR TITLE
[WIP] Github action to apply patterns

### LINKED ISSUE #15: GitHub Action for Service Mesh Patterns
#### Current Behavior
There are a couple GitHub Actions for Meshery CLI interactions currently, but not a GitHub Action specifically for service mesh patterns.

#### Desired Behavior
 It would be nice to have a GitHub Action dedicated to use cases of importing and applying service mesh patterns.

#### Implementation
Build a new action here - https://github.com/layer5io/mesheryctl-service-mesh-patterns-action

#### Acceptance Tests
1. Users can import service mesh patterns.
1. Users can apply service mesh patterns.

---
**[Contributor Guide](https://docs.meshery.io/project/contributing)**

### PR DESCRIPTION
**Pull request by hershd23**
**Description**

This PR fixes #15

**Notes for Reviewers**

**[Signed commits](../CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.
 
TO:DOs
* The `pattern_actions.yml` points to a branch on a local fork of `meshery-service-mesh-patterns-action`. This has to be changed back to the master branch of the actual repo. A PR will be raised for `meshery-service-mesh-patterns-action` as well.
* Allow users to add their own kube config to the workflow.

Question :- @leecalcote @utkarsh-pro @theBeginner86
There are presently 2 ways for getting a pattern file
1. URL to the file
2.  Add it to the repo and the path to be added is relative to the `.github` directory.

I was thinking of doing 2 for the kube config. Let me know what you think.

Also kube config has some values that are local to my system like path to certificates and client-key. Is there a way to get rid of these and import/export rest of the config?

### CODE REVIEW

**Review by iambami — APPROVED** [review]

**Review by leecalcote — APPROVED** [review]
Let's give this a shot, @hershd23

**Comment by leecalcote** [comment]
@hershd23 the client key is necessary for Meshery to auth to K8s.

In terms of where to place the kubeconfig, for now, let's require this as input into the manually executed workflow.

**Review by Chadha93 — DISMISSED** [review]
Great demo, if any blockers/requirements you can ask in Meshery slack channel :)

**Comment by Chadha93** [comment]
@hershd23 Is this PR good to go?

File: .github/workflows/pattern-action.yml [hunk 1]
```diff
@@ -1,23 +1,35 @@
...
-        uses: sayantan1413/mesheryctl-service-mesh-patterns-action@master
+      - name: Setup Kubernetes
+        uses: manusa/actions-setup-minikube@v2.4.1
+        with:
+          minikube version: 'v1.23.2'
+          kubernetes version: 'v1.22.2'
+          driver: docker
+
+      - name: Apply Pattern
+        uses: hershd23/meshery-service-mesh-patterns-action@pattern-import-apply-ci
```
Comment (hershd23) [inline .github/workflows/pattern-action.yml:31]: @Chadha93 I guess this line needs to be reverted back to the base repo. Will do that and commit

**Comment by l5io** [comment]
🚀 Preview for commit da74568f079e3e1c9966a460f8dec202995f8a4b at: https://643797d7d4d2c94edcc8cd64--service-mesh-patterns.netlify.app

**Comment by l5io** [comment]
🚀 Preview for commit b12ca17824b99c915cc89627a879f175d1787152 at: https://64379939f1730b57c09c0944--service-mesh-patterns.netlify.app

**Pull request closed without merging**